### PR TITLE
Backport the env var parser

### PR DESCRIPTION
**Pull request by afnanenayet**
## PR Summary

This PR backports the env var parser to v2, see #110 for more details.

 - **Type:** Improvements
 - **Issue Link:** https://github.com/blend/go-sdk/issues/??
 - **Intended Change Level:** minor

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

**Comment by afnanenayet** [comment]
@alexfish8

**Comment by afnanenayet** [comment]
@wcharczuk bump

**Comment by wcharczuk** [comment]
@afnanenayet before the bump, generally want to make sure the branch is updated and tests are passing

File: env/parse.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
+package env
+
+import (
+	"fmt"
+	"unicode"
+
+	"strings"
+
+	"bytes"
+	ex "github.com/blend/go-sdk/exception"
```
Comment (dhermes) [inline env/parse.go]: Alias to preserve similarity with v3?

File: env/parse.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
+package env
```
Comment (dhermes) [inline env/parse.go]: This file is significantly different from https://github.com/blend/go-sdk/blob/6051c0b95d8d63733ac88802789a93a72b87ec07/env/parse.go, I expected the diff to be very small. Is there a reason why it isn't?

**Comment by wcharczuk** [comment]
can re-open later, clearing out PRs

**Pull request closed without merging**